var/spack/repos/builtin/packages/netlib-lapack/package.py

```diff
@@ -42,10 +42,13 @@ class NetlibLapack(CMakePackage):
     variant('xblas', default=False,
             description='Builds extended precision routines using XBLAS')
 
-    patch('ibm-xl.patch', when='@3.7: %xl')
-    patch('ibm-xl.patch', when='@3.7: %xl_r')
+    patch('ibm-xl.patch', when='@3.7:3.8.9999 %xl')
+    patch('ibm-xl.patch', when='@3.7:3.8.9999 %xl_r')
```

Comment:
Spack version semantics are quite tricky, so it's hard to remember that the above is equivalent to the simpler and less arbitrary:
```suggestion
    patch('ibm-xl.patch', when='@3.7:3.8 %xl')
    patch('ibm-xl.patch', when='@3.7:3.8 %xl_r')
```

To test the version specification (thanks to @tgamblin for teaching me this trick):
```console
$ spack-python
Spack version 0.16.2
Python 3.8.5, Darwin x86_64
>>> from spack.version import ver
>>> ver('@3.8.999') in ver('@3.7:3.8')
```

Reply:
> Spack version semantics are quite tricky, so it's hard to remember that the above is equivalent to the simpler and less arbitrary:
> 
> To test the version specification (thanks to @tgamblin for teaching me this trick):
> 
> ```
> $ spack-python
> Spack version 0.16.2
> Python 3.8.5, Darwin x86_64
> >>> from spack.version import ver
> >>> ver('@3.8.999') in ver('@3.7:3.8')
> ```

I'm happy to use the simpler syntax. I only used the "four-nines" syntax because it's used elsewhere in the `netlib-lapack` recipe.